src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VideoFanTakeover.java

```diff
@@ -100,24 +104,24 @@ public void clickOnAdImage(){
   }
 
   public void clickOnVideoCloseButton() {
-    wait.forElementVisible(By.cssSelector(String.format(UI_ELEMENT_CLOSE_BUTTON_FORMAT, slotName))).click();
+    scrollToAdsElement(videoCloseButton(slotName), 60);
+
+    videoCloseButton(slotName).click();
     PageObjectLogging.log("clickOnVideoCloseButton", "close video button clicked", true, driver);
   }
 
   public Double getCurrentVideoTimeOnDesktop() {
-    String result;
-
-    driver.switchTo().frame(driver.findElement(By.cssSelector(String.format(VIDEO_IFRAME_SELECTOR_FORMAT, slotName))));
-    result = driver.findElement(By.cssSelector("video")).getAttribute("currentTime");
-    driver.switchTo().defaultContent();
-
-    return Double.parseDouble(result);
+    return getCurrentVideoTime(VIDEO_IFRAME_SELECTOR_FORMAT);
   }
 
   public Double getCurrentVideoTimeOnFandom() {
+    return getCurrentVideoTime(VIDEO_IFRAME_SELECTOR_FANDOM_FORMAT);
+  }
+
+  private Double getCurrentVideoTime(String selectorFormat) {
```

Comment:
on the other hand this method is private and is used by either `getCurrentVideoTimeOnFandom` and `getCurrentVideoTImeOnDesktop`. Those methods are public and know about the proper selector format.